dask/dataframe/tests/test_indexing.py

```diff
@@ -34,13 +34,17 @@ def test_loc():
     assert_eq(d.loc[3:], full.loc[3:])
     assert_eq(d.loc[[5]], full.loc[[5]])
     assert_eq(d.loc[[3, 4, 1, 8]], full.loc[[3, 4, 1, 8]])
+    assert_eq(d.loc[[3, 4, 1, 9]], full.loc[[3, 4, 1, 9]])
+    assert_eq(d.loc[np.array([3, 4, 1, 9])], full.loc[np.array([3, 4, 1, 9])])
```

Comment:
I don't think this works because if the `parts` dictionary is empty, then so will the `dsk` dictionary. This creates an error in the form `KeyError: ('loc-fac1d78b42c178a24a90b82fdd82445d', 0)` later. There might be an obvious fix here, but I'm not terribly familiar with the internals of dask dataframe.